spreadsheet-weather-based-marketing/main-INOUT.gs

```diff
@@ -0,0 +1,77 @@
+function main(onlyInQueue = false) {
+    // If the function is triggered by the standard trigger, it receives
+    // the trigger info object as a first param.
+    if (typeof onlyInQueue !== "boolean") {
+        onlyInQueue = false;
+    }
+
+    // Register sheet processors
+    Strategy.register('IN', config.get('col-api-url'), INAnyAPI);
+    //Strategy.register('IN', config.get('col-lat'), OpenWeatherAPI);
+    //Strategy.register('OUT', config.get('col-advertiser-id'), DV360API);
+
+    sheetsApi.getSheetObject();
```

Comment:
Hi,
 perhaps it would be better to extract the logic of processing the spreadsheet provided config into a separate method / class. 
This will simplify the loop and the retries 
Writing back of the results back to the spreadsheet, should also be handled separately. 

to simplify this function i can imagine it just going through steps as described here. 

Create config object 

for each row : 
        strategy.process("IN", config ... )
        strategy.process("OUT"... )
        sheet.update(results) 
        
        // retry if unsuccessful 

end foreach